src/mapblock.cpp

```diff
@@ -489,7 +464,7 @@ void MapBlock::deSerialize(std::istream &in_compressed, u8 version, bool disk)
 
 	u8 flags = readU8(is);
 	is_underground = (flags & 0x01) != 0;
-	m_day_night_differs = (flags & 0x02) != 0;
+	m_is_air = (flags & 0x02) == 0;
```

Comment:
This is for the !disk case.
The idea is that in the future clients can use this flag - assuming new enough server code.

Reply:
but how does the client know that it can rely on the value? we're not bumping any version numbers so this doesn't work in practice

Reply:
It can't. Servers change faster than protocol versions, though.
Right now it is not even used at the client, and I want to make sure we won't forget about this.

What is the alternative? Set `m_is_air_expired` in the `!disk` case?

Reply:
> Set `m_is_air_expired` in the `!disk` case?

I'd rather play it safe and do that, yes.

Reply:
Can I vary the Client-Server protocol without changing the version on disk?
The client and server negotiate their mutual version, and so that would work.

Reply:
Looking at the code I can increase `SER_FMT_VER_HIGHEST_READ` and then read that bit if either `disk` is true, or the the server supports the latest version.
I'll add a commit for that, and then you can have a look.

Reply:
@sfan5 see latest commit.

Client and server can now negotiate a serialized version of 30, in which case the client can trust that bit.
Max on disk version remains at 29.

Reply:
There's no technical problem with this but IMHO we shouldn't introduce this weird version negotiation. The client doesn't even have an use for the flag yet.
I'd rather someone someday does a proper version bump to 30 once the flag is actually needed.

Reply:
Happy to change it.
I just fear we'll forget about it and then introduce unnecessary work on an already slow client.
Perhaps we can somehow disallow calling `is_air` on the client. Or perhaps I am over-analyzing this :)

Reply:
I would:
* set air to expired if `!disk`
* add a comment right next to the flag reading code: `// IMPORTANT: when the version is bumped to 30 we can read m_is_air from here`
* ???
* profit

Reply:
done

Reply:
my suggestion includes reverting the 'version 30' thing, to be clear

Reply:
Oops. Yes. Forgot to add that to the commit.

Reply:
Done now. Sorry. (I was doing this in another repo and then copied changed files over - since this was not changed from what was checked in, I missed.)

Reply:
@sfan5 How is this?

Reply:
Looks good. I'll try to give this a proper test this evening.